main/fmt.c

```diff
@@ -73,7 +73,7 @@ static int printTagField (fmtSpec* fspec, MIO* fp, const tagEntryInfo * tag)
 				break;
 		}
 
-		if (findex == tag->usedParserFields)
+		if (!f)
```

Comment:
shoot, sorry, this actually isn't equivalent: here `f` won't be `NULL` unless `tag->usedParserFields == 0`; but before it also caught the case when no field passed the `isParserFieldCompatibleWithFtype (f, ftype)` test.

So this should be fixed, maybe like:
```C
 		for (findex = 0; findex < tag->usedParserFields; findex++)
 		{
            tagField *fitem = getParserField(tag, findex);
 			if (isParserFieldCompatibleWithFtype (fitem, ftype))
            {
 				f = fitem;
                break;
            }
 		}
```
or
```C
 		for (findex = 0; findex < tag->usedParserFields; findex++)
 		{
            f = getParserField(tag, findex);
 			if (isParserFieldCompatibleWithFtype (f, ftype))
                break;
            else
 				f = NULL;
 		}
```

Reply:
OK so following @masatake's comment, I'd do something like this:
```C
		for (unsigned int findex = 0; ! field && findex < tag->usedParserFields; findex++)
		{
			tagField *f = getParserField(tag, findex);
			if (isParserFieldCompatibleWithFtype (f, ftype))
 				field = f;
 		}
```
(replacing `f` with `field` for clarity in the function body, and stopping use of `findex` outside the loop body)

Reply:
Agree - then you guys take care of the right if condition and the other fix I would presume.